test/unit/pool/test-destroy-connection-timeout.js

```diff
@@ -0,0 +1,25 @@
+
+var assert     = require('assert');
+var common     = require('../../common');
+var Connection = common.Connection;
+var pool       = common.createPool({port: common.fakeServerPort});
+
+var server = common.createFakeServer();
+
+server.listen(common.fakeServerPort, function (err) {
+  assert.ifError(err);
+  pool.getConnection(function (err, connection) {
+    assert.ifError(err);
+    process.nextTick(function() {
+      connection.destroy();
+      pool.end(function (err) {
+        assert.ifError(err);
+        server.destroy();
+      })
+    })
+  })
+});
+
+setTimeout(function () {
+  throw new Error("PoolConnection.destroy left timers active after teardown")
+}, 5000).unref()
```

Comment:
> Can you achieve the goal of this test without the unref?

I'll try using the same `Timers` api that this module uses internally.

An alternative approach, that doesn't use timeouts, would be to assert that `connection.destroy()` didn't enqueue anything.  Maybe using something like:

```
connection.once("enqueue", function() { throw new Error() })
connection.destroy()
```

But that still feels like it's testing implementation details rather than user-visible behaviour.